kolibri/plugins/user_profile/viewsets.py

```diff
@@ -70,3 +73,20 @@ def post(self, request, *args, **kwargs):
                 return Response({"error": response.json()["detail"]})
         except Exception as e:
             raise ValidationError(detail=str(e))
+
+
+class LoginMergedUserViewset(APIView):
+    """
+    Viewset to login into kolibri using the merged user,
+    after the old user has been deleted
+    """
+
+    def post(self, request):
+        username = request.data.get("username", None)
+        facility = request.data.get("facility", None)
+        token = request.data.get("token", None)
+        if not TokenGenerator().check_token(username, token):
```

Comment:
Could do the lookup before the token check to still use the user pk like the base implementation does.

The advantage here is that then it's a token for the uuid rather than the username, so it's completely unique to the user (whereas by username is more generic).

Reply:
good idea, done!